K8SSAND-870 ⁃ IPCControllerTest fails when run on MacOS

When trying to run IPCControllerTest on MacOS, the test fails with:


```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.447 s <<< FAILURE! - in com.datastax.mgmtapi.ipc.IPCControllerTest
[ERROR] clientServerTest(com.datastax.mgmtapi.ipc.IPCControllerTest)  Time elapsed: 0.414 s  <<< ERROR!
io.netty.channel.ChannelException: Unable to create Channel from class class io.netty.channel.epoll.EpollServerDomainSocketChannel
    at com.datastax.mgmtapi.ipc.IPCControllerTest.clientServerTest(IPCControllerTest.java:123)
Caused by: java.lang.reflect.InvocationTargetException
    at com.datastax.mgmtapi.ipc.IPCControllerTest.clientServerTest(IPCControllerTest.java:123)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class io.netty.channel.epoll.Native
    at com.datastax.mgmtapi.ipc.IPCControllerTest.clientServerTest(IPCControllerTest.java:123)
```

Epoll is not available on MacOS, so it should try to use NIO in this case.



┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-870) by [Unito](https://www.unito.io)
┆Issue Number: K8SSAND-870
┆Priority: Medium

--- a/management-api-common/src/main/java/com/datastax/mgmtapi/ipc/IPCController.java
+++ b/management-api-common/src/main/java/com/datastax/mgmtapi/ipc/IPCController.java
@@ -25,6 +25,9 @@ import io.netty.channel.epoll.Epoll;
 import io.netty.channel.epoll.EpollDomainSocketChannel;
 import io.netty.channel.epoll.EpollEventLoopGroup;
 import io.netty.channel.epoll.EpollServerDomainSocketChannel;
+import io.netty.channel.kqueue.KQueue;
+import io.netty.channel.kqueue.KQueueDomainSocketChannel;
+import io.netty.channel.kqueue.KQueueServerDomainSocketChannel;
 import io.netty.channel.unix.DomainSocketAddress;
 
 /**
@@ -74,11 +77,33 @@ public class IPCController
 
         if (isClient)
         {
-            b = b.channel(EpollDomainSocketChannel.class);
+            if (Epoll.isAvailable())
+            {
+                b = b.channel(EpollDomainSocketChannel.class);
+            }
+            else if (KQueue.isAvailable())
+            {
+                b = b.channel(KQueueDomainSocketChannel.class);
+            }
+            else
+            {
+                throw new RuntimeException("Neither epoll nor kqueue was found. Unable to initialize channel pipeline");
+            }
         }
         else
         {
-            b = b.channel(EpollServerDomainSocketChannel.class);
+            if (Epoll.isAvailable())
+            {
+                b = b.channel(EpollServerDomainSocketChannel.class);
+            }
+            else if (KQueue.isAvailable())
+            {
+                b = b.channel(KQueueServerDomainSocketChannel.class);
+            }
+            else
+            {
+                throw new RuntimeException("Neither epoll nor kqueue was found. Unable to initialize server channel pipeline");
+            }
         }
 
         return isClient ? b.handler(channelPipeline) : ((ServerBootstrap)b).childHandler(channelPipeline);

